### PR TITLE
feat(server): add isItemReferenced query

### PR DESCRIPTION
**Pull request by nourbalaha**
# Overview
- Add `isItemReferenced` query to check if an item is referenced by any other item

### CODE REVIEW

File: server/internal/usecase/interactor/item.go [hunk 1]
```diff
@@ -160,6 +160,31 @@ func (i Item) Search(ctx context.Context, q *item.Query, sort *usecasex.Sort, p
 	return i.repos.Item.Search(ctx, q, sort, p)
 }
 
+func (i Item) CheckIfItemIsReferenced(ctx context.Context, itemId id.ItemID, correspondingFieldId id.FieldID, _ *usecase.Operator) (*bool, error) {
```
Comment (rot1024) [inline server/internal/usecase/interactor/item.go:163]:
```suggestion
func (i Item) CheckIfItemIsReferenced(ctx context.Context, itemId id.ItemID, correspondingFieldID id.FieldID, _ *usecase.Operator) (bool, error) {
```

There is reason to use a pointer bool

File: server/schemas/item.graphql [hunk 2]
```diff
@@ -125,6 +125,7 @@ extend type Query {
     sort: ItemSort
     pagination: Pagination
   ): ItemConnection!
+  checkIfItemIsReferenced(itemId: ID!, correspondingFieldId: ID!): Boolean
```
Comment (rot1024) [inline server/schemas/item.graphql:128]:
```suggestion
  checkIfItemIsReferenced(itemId: ID!, correspondingFieldId: ID!): Boolean!
```

Comment (nourbalaha) [reply server/schemas/item.graphql:128]: Done!

File: server/internal/usecase/interactor/item.go (hunk 1 above)
Comment (nourbalaha) [reply server/internal/usecase/interactor/item.go:163]: I wanted to return nil in case of an error, I will return false.
Done!

**Review by rot1024 — APPROVED** [review]

**Pull request merged**